server/devpi_server/extpypi.py

```diff
@@ -416,7 +416,10 @@ def get_simplelinks_perstage(self, project):
         # Returning a 200 with "no such project" was originally meant to
         # provide earlier versions of easy_install/pip to request the full
         # simple page.
-        serial = int(response.headers.get(str("X-PYPI-LAST-SERIAL"), "-1"))
+        try:
+            serial = int(response.headers.get(str("X-PYPI-LAST-SERIAL"), "-1"))
```

Comment:
I'd remove the "-1". That way the code path is the same way for any missing or invalid header combination.